Services/UI/OJS.Services.Ui.Data/Implementations/ContestsDataService.cs

```diff
@@ -195,6 +196,26 @@ public Task<bool> IsUserInExamGroupByContestAndUser(int id, string userId)
                 .ThenBy(c => c.PracticeStartTime);
         }
 
+        if (sorting == ContestSortType.OrderBy)
+        {
+            switch (categoriesCount)
+            {
+                case 0:
+                    return contests
+                        .OrderByDescending(c => c.EndTime);
+                case 1:
+                    return contests
+                            .OrderBy(c => c.OrderBy)
```

Comment:
I would put a comment in the code with explanation why we count categories here and why this should affect the order when `sorting == ContestSortType.OrderBy`. Straight away it looks weird compared to the previous sorting's logic.

Reply:
Added comments, commented out other sortings (no longer needed but were missed) and fixed the default sorting - first to order by the category's orderby and them by the contest's orderby